source/language/openpulse.rst

```diff
@@ -0,0 +1,317 @@
+as a first argument, since pulses need to have a definite length. Using the
+hardware dependent ``dt`` unit is recommended, since the compiler may need to
+down-sample a higher precision pulse to physically realize it.
+
+.. code-block:: none
+
+   // amp is pulse amplitude at center
+   // center is the mean of pulse
+   // sigma is the standard deviation of pulse
+   gaussian(length:l, complex[float[32]]:amp, length:center, length:sigma)
+
+   // amp is pulse amplitude at center
+   // center is the mean of pulse
+   // sigma is the standard deviation of pulse
+   sech(length:l, complex[float[32]]:amp, length:center, length:sigma)
+
+   // amp is pulse amplitude at center
+   // center is the mean of pulse
+   // square_width is the width of the square pulse component
+   // sigma is the standard deviation of pulse
+   gaussian_square(length:l, complex[float[32]]:amp, length:center, length:square_width, length:sigma)
+
+   // amp is pulse amplitude at center
+   // center is the mean of pulse
+   // sigma is the standard deviation of pulse
+   // beta is the Y correction amplitude, see the DRAG paper
+   drag(length:l, complex[float[32]]:amp, length:center, length:sigma, float[32]:beta)
+
+   // Define a constant pulse of length l
+   constant(l:length)
+
+Play instruction
+----------------
+
+Pulses are scheduled using the ``play`` instruction. These instructions may
+only appear inside a ``defcal`` block!
+
+Play instructions have two required parameters:
+
+- a value of type ``pulse`` representing the pulse envelope
+- the frame to use for the pulse
+
+.. code-block:: none
+
+   // Play a 3 sample pulse on qubit 0's "drive" frame
+   play([1+0*j, 0+1*j, 1/sqrt(2)+1/sqrt(2)*j]) frameof("drive", %0);
+
+   // Play a gaussian on qubit 1's "drive" frame
+   frame f1 = frameof("drive", %1);
+   play(gaussian(...)) f1;
+
+Capture Instruction
+-------------------
+
+Acquisition is scheduled by a ``capture`` instruction. This is a special
+``kernel`` function which is specified by a hardware vendor. The measurement
+process is difficult to describe generically due to the wide variety of
+hardware and measurement methods. Like the play instruction, these instructions
+may only appear inside a ``defcal`` block!
+
+The only required parameter is a ``frame``.
+
+The following are possible parameters that might be included:
+
+- A "duration" of type ``length``, if it cannot be inferred from other parameters
+- A "filter", which is dot product-ed with the measured IQ the distill the
+  result into a single IQ value
+- A "tag", which could be used to identify which branch of an if statement was
+  traversed
+
+Again it is up to the hardware vendor to determine the parameters and write a
+kernel definition at the top-level, such as:
+
+.. code-block:: none
+
+   // Minimum requirement
+   kernel capture(frame output) -> complex[32];
+
+   // A capture command with more features
+   kernel capture(frame output, pulse filter) -> complex[32];
```

Comment:
Maybe it's just me, but when I first look at this the terms `output` and `filter` look confusing because  my inclination is to assume they are qualifiers. I think it would be more instructive to :fire: those additional terms and add an explicit example.